### PR TITLE
Feature/interactive table directive

### PR DESCRIPTION
**Pull request by ulferts**
Extract the dynamic width calculation from the work-packages-table-directive into it's own interactive-table-directive.

The directive is then applied to the work-packages table so the functionality remains the same.

Additionally, this PR removes the proxy styles <sup>TM</sup> for work packages that where there just because hadn't been adapted to use the `generic-table--...` classes yet.

I would have loved to do more and actually clean up the directive code, but that was out of scope.

https://community.openproject.org/work_packages/21235

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://ulferts-feature-interactive-table-directive.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://ulferts-feature-interactive-table-directive.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

**Comment by oliverguenther** [comment]
I've just applied this to my changes for testing purposes and it works splendidly.

File: app/assets/stylesheets/content/_work_packages_table.lsg [hunk 1]
```diff
@@ -3,9 +3,9 @@
 ## with work packages
 
 ```
-<div class="work-packages-table--container">
-  <div class="work-packages-table--results-container">
-    <table class="workpackages-table">
+<div class="generic-table--container work-package-table--container" interactive-table>
```
Comment (NobodysNightmare) [inline app/assets/stylesheets/content/_work_packages_table.lsg:6]: Just a question of repetitiveness:

Do we still need the WP table component in the LSG?

Since you removed the `work-packages-table` class and this is now also a `generic-table`, I would have assumed that one example for this kind of table is enough.

You might want to use work packages as an _example_ for such a table, though... (instead of the lorem ipsum)

File: app/assets/stylesheets/content/_work_packages_table.sass [hunk 2]
```diff
@@ -25,41 +25,3 @@
 //
 // See doc/COPYRIGHT.rdoc for more details.
 //++
-
-@import 'table'
```
Comment (NobodysNightmare) [inline app/assets/stylesheets/content/_work_packages_table.sass:29]: image

WHY U NO DELETE WHOLE FILE?

**Comment by NobodysNightmare** [comment]
Oh...

Reading this I remember that this is the code I already saw when profiling the work packages table.

If you ever wondered what was taking up so much of the render time for the WP table: You just extracted it into its own directive. Name choice makes this somewhat ironical :smile: 

However: Since you just seem to have moved the code I will ignore the that fact for now.

**Comment by NobodysNightmare** [comment]
Not sure why, but in this PR I can see a visual glitch that is not present on community:

column-glitch

For the "check" column, the heading seems to be too large when it is highlighted...

**Comment by NobodysNightmare** [comment]
**tl; dr:**
- [ ] deduplicate LSG
- [ ] remove empty file
- [ ] fix "check" header
- [ ] get travis green

File: app/assets/stylesheets/content/_work_packages_table.sass (hunk 2 above)
Comment (ulferts) [inline app/assets/stylesheets/content/_work_packages_table.sass:29]: It's because I know more than you do. 

@HDinger does alter the behaviour (min-width) of the columns again in https://github.com/HDinger/openproject/blob/feature/21231-apply-table-styling/app/assets/stylesheets/content/_work_packages_table.sass#L41

I therefore didn't want her to have to recreate the file again.

**Comment by oliverguenther** [comment]
https://github.com/ulferts/openproject/pull/3 should address all broken specs and cukes.

File: app/assets/stylesheets/content/_work_packages_table.lsg (hunk 1 above)
Comment (ulferts) [inline app/assets/stylesheets/content/_work_packages_table.lsg:6]: I think we are not there yet, when it comes to being able to remove the work packages table from the LSG. 

This table still has some unique capabilities (although they do not work in the LSG): column menu, row menu, ...

But until we are able to have the same capabilities for all the tables, I would leave it there.

**Comment by oliverguenther** [comment]
> @NobodysNightmare 
> Not sure why, but in this PR I can see a visual glitch that is not present on community:
> 
> For the "check" column, the heading seems to be too large when it is highlighted...

The width of the checkbox is not computed as the element is not matched in the directive (it matches header, not header-outer). I can't seem to find references to `generic-table--header` itself, so I suppose the following change is enough (it fixes the issue here)

``` diff
diff --git a/frontend/app/ui_components/interactive-table-directive.js b/frontend/app/ui_components/interactive-table-directive.js
index b2d2169..f129043 100644
--- a/frontend/app/ui_components/interactive-table-directive.js
+++ b/frontend/app/ui_components/interactive-table-directive.js
@@ -47,7 +47,7 @@ module.exports = function($timeout, $window){
       function getHeadersFooters() {
         return element.find(
           '.generic-table--sort-header-outer,' +
-          '.generic-table--header,' +
+          '.generic-table--header-outer,' +
           '.generic-table--footer-outer'
         );
       }

```

**Comment by oliverguenther** [comment]
Specs seem fine except for that one rule I've missed over at https://github.com/ulferts/openproject/pull/4

**Comment by NobodysNightmare** [comment]
LGTM, lets wait for travis again then ^^

:+1:
